### ISSUE TITLE
List of minor modes in tooltip is always for selected window

### ISSUE THREAD

**Issue by cpitclaudel** [Malabarba/rich-minority#13]
This could be an SML issue. To reproduce, open two buffers, enable e.g. outline-minor-mode in only one of them, and select the corresponding window.  Then the `Full list:` help-echo of the minor mode section of the mode line will always mention outline-minor-mode, though it's only enabled in one of the buffers.

In pictures:

* Fundamental mode buffer selected:

![screenshot from 2016-07-01 15 47 50](https://cloud.githubusercontent.com/assets/8181630/16532928/412f9d36-3fa3-11e6-8cd1-e17e46924647.png)

* Prog mode buffer selected:

![screenshot from 2016-07-01 15 47 59](https://cloud.githubusercontent.com/assets/8181630/16532935/4b4fcb1a-3fa3-11e6-96b1-70347e26001f.png)
![screenshot from 2016-07-01 15 48 05](https://cloud.githubusercontent.com/assets/8181630/16532936/4b536ee6-3fa3-11e6-9ded-53682b3dcaa7.png)

Notice how the help echo is always that of the selected window. Hopefully this report makes sense :/

**Comment by Malabarba** [comment]
Yeah, I've noticed that too. We probabley need to add a `(with-current-buffer (window-buffer (selected-window)) ...)` in there.

**Comment by Malabarba** [comment]
Nope, that didn't work. I'm playing around with this, but I can't seem to find a good solution.
